server/controllers/UsersController.js

```diff
@@ -98,6 +111,92 @@ class UsersController {
         }
       }));
   }
+
+  /**
+   * This controller updates a users profile
+   * @param {object} req
+   * @param {object} res
+   * @param {object} next
+   * @returns {void}
+   */
+  static updateProfile(req, res, next) {
+    // get token from request header
+    const token = req.headers.authorization;
+    if (!token) {
+      const error = new Error('no token provided');
+      error.status = 401;
+      return next(error);
+    }
+
+    // check if request token id matches id of account to be updated
+    const { userId } = req.params;
+    const { id } = verifyToken(token);
+    if (id !== parseInt(userId, 10)) {
+      const error = new Error('can\'t update another user\'s profile');
+      error.status = 403;
+      return next(error);
+    }
+
+    // check that the req body contains only valid columns
+    const updateData = req.body;
+    const profileFields = [
+      'fullName',
+      'email',
+      'bio',
+      'avatarUrl',
+      'location',
+      'facebookUrl',
+      'twitterUrl',
+    ];
+    const updateFields = Object.keys(updateData);
+    updateFields.forEach((field) => {
+      if (profileFields.indexOf(field) < 0) {
+        const error = new Error(`user column '${field}' does not exist`);
+        error.status = 422;
+        throw error;
+      }
+    });
+
+    const updateProfile = () => {
+      User.update(updateData, {
+        where: {
+          id: {
+            [Op.eq]: userId
+          }
+        }
+      })
+        .then((rowCount) => {
+          if (rowCount[0] > 0) {
+            return res.status(200).json({
+              status: 'success',
+              message: 'one user profile updated successfully',
+              rowCount: rowCount[0],
+            });
+          }
+          const error = new Error('update unsuccessful, user not found');
+          error.status = 404;
+          throw error;
+        })
+        .catch((err) => {
+          next(err);
+        });
+    };
+
+    if (req.file) {
+      winston.log('info', 'uploading image to cloud...');
+      cloudinary.v2.uploader.upload(req.file.path, (err, result) => {
+        if (err) {
+          winston.log('info', 'upload failed');
+          next(err);
+        }
+        winston.log('info', 'upload complete');
```

Comment:
Nice. Shouldn't winston only be used for development purposes?

Reply:
Yeah it is, this will just let us know when the image is uploading and completes uploading. Cause it takes a while to upload

Reply:
Why should `winston` be on `Dev` only @KvNGCzA. #seekingClarification

Reply:
@mbilesanmi because it does the same thing as console.log and I assumed that once you are done testing you should remove it.